p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -2647,196 +2645,191 @@ X.v1  // reference to v1
+  same type as the operands. It is computed by adding the negation
   of the second operand (similar to C).
-- Multiplication ```*```. Result has the same width as the
-  operands. P4 targets may impose additional restrictions (e.g., may
-  only allow multiplications with powers of two).
-- Bitwise "and" between two bit-strings of the same width, designated
-  by ```&```
-- Bitwise "or" between two bit-strings of the same width, designated
-  by ```|```
-- Bitwise "complement" of a single bit-string, designated by ```~```
-- Bitwise "xor" of two bit-strings of the same width, designated by ```^```
-
-There are also the following operations:
-
-- Concatenation of two bit-strings, resulting in a bit-string whose
-  length is the sum of the lengths, designated by the infix operator ```++```.
-  The left bit-string provides the most significant bits.
-- Extraction of a set of contiguous bits (bit slice), designated by ```[m:l]```,
+- Multiplication, denoted by ```*```. The result has the same width as the
+  operands and is computed by truncating the result to the width of the output.
+  P4 architectures may impose additional restrictions---e.g., they may
+  only allow multiplication by a power of two.
+- Bitwise "and" between two bit-strings of the same width, denoted
+  by ```&```.
+- Bitwise "or" between two bit-strings of the same width, denoted
+  by ```|```.
+- Bitwise "complement" of a single bit-string, denoted by ```~```.
+- Bitwise "xor" of two bit-strings of the same width, denoted by ```^```.
+
+Bit-strings also support the following operations:
+
+- Concatenation of two bit-strings, denoted by the infix operator ```++```. The result is a bit-string whose
+  length is the sum of the lengths of the inputs where the most significant bits are taken from the left operand.
+- Extraction of a set of contiguous bits, also known as a slice, denoted by ```[m:l]```,
   where ```m``` and ```l``` must be positive integers
   that are compile-time known values, and ```m >= l```. The result is
   a bit-string of width ```m - l + 1```, including the bits numbered
-  from ```l``` (which becomes the LSB of the result) to ```m``` (the
-  MSB of the result) from the source operand. The conditions ```0 <= l < W```
+  from ```l``` (which becomes the least significant bit of the result) to ```m``` (the
+  most significant bit of the result) from the source operand. The conditions ```0 <= l < W```
   and ```l <= m < W``` are checked statically (where ```W``` is
   the length of the source bit-string). Note that both endpoints of
   the extraction are inclusive. The bounds are required to be
   compile-time known values so that the result width can be computed
-  at compile time.
-- Slices are l-values: one can assign to a slice: ``` e[m:l] = x ```
-  This statement sets bits ```m``` to ```l``` of ```e``` to the
+  at compile time. Slices are also l-values, which means taht P4 supports asssigning to a slice: ``` e[m:l] = x ```.
```

Comment:
Ha. I had the kids from wake-up until mid-afternoon so she could go ride up Montebello with her friends (and sacrificed "my" exercise window to do a bit of polishing).